rope/refactor/move.py

```diff
@@ -324,15 +324,21 @@ def get_changes(
           will be applied to all python files.
 
         """
+        # To do (in another PR): Create a new var: dest_resource: resource.Reource
```

Comment:
I wouldn't really recommend creating a new argument here. get_changes() is a public API, pretty much all integrations that supports refactoring would have directly called it. I would be very cautious on changing the API here as it's likely going to break some integrations.

What I'd recommend here is to annotate `is_folder()` to become a [TypeGuard](https://docs.python.org/3/library/typing.html#typing.TypeGuard). Unfortunately, TypeGuard is only available in Python 3.9 and above. 

IMO, we don't need to maintain backwards compatibility with type annotation. If static checks doesn't pass in older Python version, as long as the code is unaffected at runtime, that's perfectly fine to me. 

Alternatively, we could add development dependency on [typing-extensions](https://pypi.org/project/typing-extensions/). which has a backport of `typing.TypeGuard`.

Reply:
The comment wasn't clear. I wasn't suggesting the api, I was suggesting adding a new *var* (not argument). This change would be an example of what I called "splitting" a var in another reply. mypy will complain about vars that hold values of differing types. It's no big deal whether this ever gets fixed.